docs/pipelines/yaml-schema.md

```diff
@@ -875,6 +876,83 @@ steps:
 ::: moniker-end
 
 ## Resources
+Any external service that is consumed as part of your pipeline is a resource. An example of a resource can be another CI/CD pipeline that produces artifacts (say Azure pipelines, Jenkins etc.), code repositories (GitHub, Azure Repos, Git), container image registries (ACR, Docker hub etc.).
+
+Resources in YAML represent sources of types pipelines, repositories and containers.
+
+#### Schema
+
+```yaml
+resources:
+  pipelines: [ pipeline ]  
+  repositories: [ repository ]
+  containers: [ container ]
+```
+
+### Pipeline resource
+If you have an Azure Pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. And you can also enable pipeline completion triggers.
+
+# [Schema](#tab/schema)
+
+```yaml
+resources: 
+  pipelines:
+  - pipeline: string  # identifier for the pipeline resource
+    project:  string # project for the build pipeline; optional input for current project
+    source: string  # source pipeline definition name
+    branch: string  # branch to pick the artifact, optional; defaults to all branches
+    version: string # pipeline run number to pick artifact; optional; defaults to last successfully completed run
+    trigger:     # optional; Triggers are not enabled by default.
+      branches:  
+        include: [string] # branches to consider the trigger events, optional; defaults to all branches.
+        exclude: [string] # branches to discard the trigger events, optional; defaults to none. 
+```
+# [Example](#tab/example)
+
+```yaml
+resources: 
+  pipelines:
+  - pipeline: MyAppA 
+    source: MyCIPipelineA
+  - pipeline: MyAppB
+    source: MyCIPipelineB
+    trigger: true
+  - pipeline: MyAppC
+    project:  DevOpsProject 
+    source: MyCIPipelineC
+    branch: releases/M159  
+    version: 20190718.2 
+    trigger:     .
+      branches:  
+        include:  
+        - master
+        - releases/*
+        exclude:   
+        - users/*  
+```
+>[!IMPORTANT]
+>When you define the resource trigger, if the `pipeline` resource is from the same repo as the current pipeline, we will follow the same branch and commit on which the event is raised. However, if the `pipeline` resource is from a different repo, the current pipeline is triggered on the master branch.
+
+#### `pipeline` resource meta-data as pre-defined variables.
```

Comment:
I'm not convinced this belongs in the schema doc. It's definitely worth putting in another topic and linking to from here, though.

Reply:
Sure. Today the resources doc (vsts-docs/docs/pipelines/process/resources.md) needs to be redone. We need to move variable groups, service connections and agent pools out of it and create a detailed concept doc about resources that talks about pipelines, repos, containers, builds and packages. As part of it I will move this section. Until then I am thinking of keeping this info here.